app/src/main/java/com/github/se/project/MainActivity.kt

```diff
@@ -40,26 +40,31 @@ class MainActivity : ComponentActivity() {
     requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
     super.onCreate(savedInstanceState)
     setContent {
-      SampleAppTheme { Surface(modifier = Modifier.fillMaxSize()) { PocketTutorApp() } }
+      SampleAppTheme {
+        Surface(modifier = Modifier.fillMaxSize()) {
+          PocketTutorApp(
+              authenticationViewModel = viewModel(),
+              listProfilesViewModel = viewModel(factory = ListProfilesViewModel.Factory),
+              lessonViewModel = viewModel(factory = LessonViewModel.Factory))
+        }
+      }
     }
   }
 }
 
 @Composable
-fun PocketTutorApp() {
+fun PocketTutorApp(
+    testMode: Boolean = false,
```

Comment:
Instead of using a argument for test mode, one other solution is to use system property defined in the build.graddle.kts that is true only for tests. This is a possible improvement to avoid hardcoding that could be changed but not necessary made now.